src/hooks/useCachedNode.ts

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useRef } from 'react'
+import { cx } from '../../styled-system/css'
+import { editableRecipe } from '../../styled-system/recipes'
+import Thought from '../@types/Thought'
+
+// Create a regular expression to match and remove BEM modifier (variant) classes
+const editableClass = cx(
+  editableRecipe({
+    preventAutoscroll: true,
+  }),
+).split(' ')
+
+// Filter classes containing '--' to identify variant classes
+const modifiers = editableClass.filter(cls => cls.includes('--'))
+
+// Precompile the regex once, so it's not recreated on each render
+const regex = new RegExp(`\\b(${modifiers.join('|')})\\b`, 'g')
+
+/**
+ * Custom hook to capture and cache the static HTML string of a node.
+ *
+ * @param thought - The thought object.
+ * @param elementRef - The ref of the node.
+ * @returns The cached HTML string.
+ */
+const useCachedNode = ({
```

Comment:
I would suggest a more specific name for this hook, e.g. `useCachedThoughtHtml`, since it has several thought-specific pieces of functionality.

We can also add to the JSDOC to explain a little more about what it does and where it's useful.